### PR TITLE
create API endpoint and remove files associated to a CT on delete

### PR DESCRIPTION
**Pull request by danigo99**
## Context

Adding functionality to delete custom type files in slice machine local.
Ticket here: https://linear.app/prismic/issue/SM-863/[delete-ct]-aauser-i-can-delete-a-custom-type-from-the-ct-list-only

## The Solution

Added API endpoint to server, and service which does the business logic

## Impact / Dependencies

This still requires some frontend work, since the button currently has no visible feedback (need to look at the network tab).
This will come form this ticket: https://linear.app/prismic/issue/SM-865/[delete-ct]-aauser-i-can-see-the-deleted-custom-types-disappear-from

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [x] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

---

image

### CODE REVIEW

File: packages/slice-machine/server/src/api/custom-types/delete.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  } catch (err) {
+    console.error(`[custom-type/delete] Could not update the project types.`);
+    return {
+      err: err,
+      reason:
+        "Something went wrong when deleting your Custom Type. Check your terminal.",
+      status: "500",
+      type: "warning",
+    };
+  }
```
Comment (danigo99) [inline packages/slice-machine/server/src/api/custom-types/delete.ts]: Should these steps all fail when one fails? Or should they all be tried and we just log the ones that didn't work? Was a bit unsure of this from the ticket

**Review by mathieud — APPROVED** [review]

File: packages/slice-machine/lib/mock/misc/fs.ts [hunk 2]
```diff
@@ -25,16 +23,13 @@ export const insert = (
...
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
   const config = getConfig(cwd);
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
   const withInsert = {
     ...config,
     ...(prefix
       ? {
           // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-assignment
           [prefix]: {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```
Comment (Heyrows) [inline packages/slice-machine/lib/mock/misc/fs.ts:37]: nice

File: packages/slice-machine/server/src/api/custom-types/delete.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import { DeleteCustomTypeQuery } from "../../../../lib/models/common/CustomType";
+import { remove as removeCtsFromMockConfig } from "../../../../lib/mock/misc/fs";
+import path, { resolve } from "path";
+
+export default async function handler(
+  req: RequestWithEnv
+): Promise<
+  { err: unknown; reason: string; status: string } | Record<string, unknown>
+> {
+  const state = await getBackendState(req.errors, req.env);
```
Comment (Heyrows) [inline packages/slice-machine/server/src/api/custom-types/delete.ts:18]: you don't need to do that, it's already in the `req: RequestWithEnv (req.env)`

File: packages/slice-machine/server/src/api/custom-types/delete.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  const { id } = req.query as DeleteCustomTypeQuery;
+  const ctFolder = CustomTypesPaths(state.env.cwd).customType(id).folder();
+  const customTypeAssetsFolder = GeneratedCustomTypesPaths(state.env.cwd)
+    .customType(id)
+    .folder();
+
+  try {
+    IO.CustomType.deleteCustomType(ctFolder);
```
Comment (Heyrows) [inline packages/slice-machine/server/src/api/custom-types/delete.ts:26]: this should be awaited, so that we don't try to do the rest if it fails.

Comment (danigo99) [reply packages/slice-machine/server/src/api/custom-types/delete.ts:26]: This function is already synchronous, so we don't need to await it right? All the other fs operations we do are the same.

**Pull request merged**